### DISCUSSION TITLE
Rewarding every node for sharing

### DISCUSSION THREAD

**Discussion by KevinDeeDev** [OpenSharedNetwork/OpenSharedNetwork#1] [Ideas]
How would you implement a reward system for network node providers?

**Comment by radioisthefuture** [comment]
I will look around for a solution and come back to you!

**Comment by radioisthefuture** [comment]
I think we need to take some sort of a blockchain/cryptocurrency route.
First thoughts:
The network in itself is decentralized so it would make sense to implement a blockchain that is hosted on the [OpenSharedNetwork](https://opensharednetwork.com/) and create a cryptocurrency where every node in the network is a validator, so every node gets credits for **handling/validating transactions/sharing bandwidth/computing power/storage.**
Something like [Monero](https://www.getmonero.org/) comes to mind but I will dig deeper to find a fitting solution.
